### PR TITLE
[ONNX Export] Use `half_pixel` instead of `pytorch_half_pixel`.

### LINKED ISSUE #79361: [ONNX Export] Interpolation likely should be exported with `half_pixel` instead of `pytorch_half_pixel`
### 🐛 Describe the bug

PyTorch currently exports Interpolation op to ONNX Resize op with `pytorch_half_pixel` mode:
https://github.com/pytorch/pytorch/blob/5413580f9e6ce1d490d0b137595574befd28d5f4/torch/onnx/symbolic_helper.py#L798

This seems to be incorrect now as it appears that the behavior of PyTorch's Interpolation has changed to `half_pixel` (see the difference to `pytorch_half_pixel` here: https://github.com/onnx/onnx/blob/main/docs/Operators.md#resize). This can be verified in the code snippet below. It compares torch output with that from ONNX's reference implementation:

```python
import torch
import numpy as np
from onnx.backend.test.case.node import resize
out_size = [1, 1]


class Model(torch.nn.Module):
    def forward(self, x):
        return torch.nn.functional.interpolate(x, size=out_size, mode='bilinear')


model = Model()
x = torch.rand([1, 1, 4, 4]).numpy()
y_torch = model(torch.from_numpy(x)).detach().numpy()
y_onnx_pytorch_hp = resize.interpolate_nd(
    x, resize.linear_coeffs, output_size=[1, 1, *out_size], coordinate_transformation_mode='pytorch_half_pixel')
y_onnx_hp = resize.interpolate_nd(
    x, resize.linear_coeffs, output_size=[1, 1, *out_size], coordinate_transformation_mode='half_pixel')

print('Comparing PyTorch to ONNX half_pixel')
np.testing.assert_allclose(
    y_torch, y_onnx_hp, err_msg='PyTorch vs half_pixel')  # succeed
print('Passed')
print('Comparing PyTorch to ONNX pytorch_half_pixel')
np.testing.assert_allclose(y_torch, y_onnx_pytorch_hp,
                           err_msg='PyTorch vs pytorch_half_pixel')  # fail
print('Passed')


torch.onnx.export(model, (torch.from_numpy(x), ),
                  "output.onnx", opset_version=14, input_names=["input"], output_names=["output"])
# output.onnx uses pytorch_half_pixel
```
Note that ONNX's reference implementation should be correct, at least it's consistent to ORT: 
```python
import onnxruntime as ort
y_ort = ort.InferenceSession("output.onnx", provider_options=[
                             "CPUExecutionProvider"]).run(["output"], {"input": x})[0]

print('Comparing ORT pytorch_half_pixel to ORT pytorch_half_pixel')
np.testing.assert_allclose(y_ort, y_onnx_pytorch_hp,
                           err_msg='ORT vs pytorch_half_pixel')  # succeed
print('Passed')
```

Morever, there is [the checking script](https://gist.github.com/daquexian/4bab98a7e706ad95616567433c1234b8) from https://github.com/onnx/onnx/pull/2057 that verifies ONNX output with PyTorch's. It now dumps a lot of errors, but dumps no error after changing `pytorch_half_pixel` to `half_pixel` (https://gist.github.com/daquexian/4bab98a7e706ad95616567433c1234b8#file-check_onnx_resize_proposal_vs_tf_and_pytorch-py-L237).

### Versions

```
Collecting environment information...
PyTorch version: 1.13.0a0+gitb651148
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 11.1.0-++20211011094159+1fdec59bffc1-1~exp1~20211011214614.8
CMake version: version 3.19.6
Libc version: glibc-2.17

Python version: 3.7.11 (default, Jul 27 2021, 14:32:16)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-90-generic-x86_64-with-debian-buster-sid
Is CUDA available: False
CUDA runtime version: 11.2.152
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2080
Nvidia driver version: 495.29.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] mypy==0.950
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.6
[pip3] onnx-pytorch==0.1.4
[pip3] torch==1.13.0a0+gitb651148
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] magma-cuda110             2.5.2                         1    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-include               2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py37h7f8727e_0  
[conda] mkl_fft                   1.3.1            py37hd3c417c_0  
[conda] mkl_random                1.2.2            py37h51133e4_0  
[conda] numpy                     1.21.6                   pypi_0    pypi
[conda] numpy-base                1.21.2           py37h79a1101_0  
[conda] onnx-pytorch              0.1.4                    pypi_0    pypi
[conda] torch                     1.13.0a0+gitb651148           dev_0    <develop>
[conda] torchvision               0.12.0                   pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by lazycal**
Fixes #79361

### CODE REVIEW

**Comment by lazycal** [comment]
cc: @justinchuby

**Comment by MohamedAliRashad** [comment]
This should be merged

**Comment by justinchuby** [comment]
Thanks for the PR! We will be reviewing it very soon

**Comment by justinchuby** [comment]
@BowenBao could you help taking a look?

**Review by BowenBao — APPROVED** [review]
LGTM, thanks for contributing!

(btw do you by any chance have the link/commit when PyTorch shifted its behavior?)

**Comment by lazycal** [comment]
Unfortunately I don't know either. I tried looking for it but didn't succeed.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -2882,6 +2882,43 @@ def forward(self, x):
     def test_interpolate_downsample(self):
         self._interpolate_tests(False)
 
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_interpolate_half_pixel(self):
+        # testing whether it uses "half_pixel" or "pytorch_half_pixel"
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:2887]: We should parameterize this test

**Pull request closed without merging**